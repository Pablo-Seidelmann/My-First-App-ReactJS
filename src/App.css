.title
{
    color: #38C98D;
    font-weight: 900px;
    font-size: 25px;
    text-align: center;
}