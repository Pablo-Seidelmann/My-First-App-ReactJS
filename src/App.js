import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import AddCategory from './components/AddCategory';
import SetCategory from './components/SetCategory';

const App = ({ defaultCategories = [] }) => {

  const [ categories, setCategories ] = useState( defaultCategories );

  const Title = styled.h2`
    color: #38C98D;
    font-weight: 900px;
    font-size: 25px;
    text-align: center;
  `

  return(

    <Fragment>

      <Title id = 'Title-App' >My First App!</Title>

      <AddCategory setCategories = { setCategories } />

      <hr />

      <ol>

        {
          categories.map( category => <SetCategory key = { category.toString() } category = { category } />)
        }
        
      </ol>

    </Fragment>

  );

};

App.propTypes = {
  Title: PropTypes.string.isRequired
}

export default App;
