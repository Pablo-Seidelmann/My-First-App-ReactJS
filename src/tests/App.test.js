import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import App from '../App';

describe('Pruebas al Componente <App />', () => {

    let wrapper;

    beforeEach(() => {

        const Title = 'My First App!';

        wrapper = shallow( <App Title = { Title } /> );

    })



    test('Snapshot al Componente Testeado', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('El titulo deberia ser el mismo', () => {
        
        const title = wrapper.find( '#Title-App' );

        expect( title.exists() ).toBe( true );

        expect( title.text().trim() ).toBe( 'My First App!' );

    });

    test('Debe de haber 2 elementos en el setCategories', () => {
       
        const Title = 'My First App!';


        const categories = ['Pablo', 'El Tula'];

        const wrapper = shallow( <App defaultCategories = { categories } Title = { Title } /> );
        
        expect( wrapper ).toMatchSnapshot();

        expect( wrapper.find( 'SetCategory' ).length ).toBe( categories.length );

    });
    
    

});
