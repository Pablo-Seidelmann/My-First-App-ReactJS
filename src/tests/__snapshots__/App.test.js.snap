// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al Componente <App /> Debe de haber 2 elementos en el setCategories 1`] = `
<Fragment>
  <styled.h2
    id="Title-App"
  >
    My First App!
  </styled.h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <SetCategory
      category="Pablo"
      key="Pablo"
    />
    <SetCategory
      category="El Tula"
      key="El Tula"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas al Componente <App /> Snapshot al Componente Testeado 1`] = `
<Fragment>
  <styled.h2
    id="Title-App"
  >
    My First App!
  </styled.h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
